src/itwinai/torch/inference.py

```diff
@@ -69,46 +78,115 @@ def __call__(self) -> nn.Module:
-    test_dataloader: DataLoader = None
+    #: PyTorch random number generator (PRNG).
+    torch_rng: torch.Generator = None
+    #: itwinai ``itwinai.Logger``
+    logger: Logger = None
 
     def __init__(
         self,
+        config: Union[Dict, TrainingConfiguration],
         model: Union[nn.Module, ModelLoader],
-        test_dataloader_class: str = 'torch.utils.data.DataLoader',
-        test_dataloader_kwargs: Optional[Dict] = None,
+        strategy: Literal["ddp", "deepspeed", "horovod"] = 'ddp',
+        logger: Optional[Logger] = None,
+        checkpoints_location: str = "checkpoints",
         name: str = None
     ) -> None:
         super().__init__(model=model, name=name)
         self.save_parameters(**self.locals2params(locals()))
-        self.model = self.model.eval()
-
-        # Train and validation dataloaders
-        self.test_dataloader_class = dynamically_import_class(
-            test_dataloader_class
-        )
-        test_dataloader_kwargs = (
-            test_dataloader_kwargs
-            if test_dataloader_kwargs is not None else {}
+        if isinstance(config, dict):
+            self.config = TrainingConfiguration(**config)
+        else:
+            self.config = config
+        self.strategy = strategy
+        self.logger = logger
+        self.checkpoints_location = checkpoints_location
+
+    @property
+    def strategy(self) -> TorchDistributedStrategy:
+        """Strategy currently in use."""
+        return self._strategy
+
+    @strategy.setter
+    def strategy(self, strategy: Union[str, TorchDistributedStrategy]) -> None:
+        if isinstance(strategy, TorchDistributedStrategy):
+            self._strategy = strategy
+        else:
+            self._strategy = self._detect_strategy(strategy)
+
+    @property
+    def device(self) -> str:
+        """Current device from distributed strategy."""
+        return self.strategy.device()
+
+    def _detect_strategy(self, strategy: str) -> TorchDistributedStrategy:
+        if not distributed_resources_available():
+            print("WARNING: falling back to non-distributed strategy.")
+            dist_str = NonDistributedStrategy()
+        elif strategy == 'ddp':
+            dist_str = TorchDDPStrategy(backend='nccl')
+        elif strategy == 'horovod':
+            dist_str = HorovodStrategy()
+        elif strategy == 'deepspeed':
+            dist_str = DeepSpeedStrategy(backend='nccl')
+        else:
+            raise NotImplementedError(
+                f"Strategy '{strategy}' is not recognized/implemented.")
+        return dist_str
+
+    def _init_distributed_strategy(self) -> None:
+        if not self.strategy.is_initialized:
+            self.strategy.init()
+
+    def distribute_model(self) -> None:
+        """
+        Distribute the torch model with the chosen strategy.
+        """
+        if self.model is None:
+            raise ValueError(
+                "self.model is None! Mandatory constructor argument "
```

Comment:
Either move this check in the constructor, or remove "constructor" from the error message. It may be a bit misleading othwerwise